torch/csrc/utils.h

```diff
@@ -81,6 +81,12 @@
 #define THPHalfUtils_unpackAccreal(object)    (double)THPUtils_unpackReal_FLOAT(object)
 #define THPHalfUtils_newAccreal(value)        THPUtils_newReal_FLOAT(value)
 
+#define THPBoolUtils_checkReal(object)        THPUtils_checkReal_INT(object)
```

Comment:
I'm not sure these are right, because most of these call e.g. PyInt_Check / PyFloat_Real but you probably want to use PyBool_Check.

Reply:
These need to stay INT as the code that is referencing these macros is expecting certain behavior. As an example, checkReal_* is being used in several places including StorageMethods.cpp in fill_() method. If we use PyBool_Check, fill_ method wont work as expected. There is a test that verifies that (it will fail in case of PyBool_Check) - test_print test in test_torch.py. Same holds for the unpackReal_* macro.

Reply:
I don't think I agree.  No existing code references this because it is new code.  The fill_() example is trying to fill a BoolStorage with an int; it's debatable what the correct behavior is here, but we can look at existing storage examples:

1) Float accepts integers or doubles
2) Int accepts only integers (not float)

Given the restrictiveness of ints, I'd make bools restrictive too.  Plus it's always easier to make something less restrictive than more restrictive (because of backwards compatibility concerns).

Note that the Tensor behavior is different than Storage, but we can just focus on Storage for now.